lib/astarte/core/generators/interface.ex

```diff
@@ -182,35 +183,53 @@ defmodule Astarte.Core.Generators.Interface do
 
   defp ownership, do: member_of([:device, :server])
 
-  defp prefix, do: MappingGenerator.endpoint()
+  defp interface_mapping(params, endpoint, postfix),
+    do: MappingGenerator.mapping(params ++ [endpoint: endpoint <> "/" <> postfix])
+
+  defp interface_mappings(:individual, params) do
+    MappingGenerator.endpoint()
+    |> bind(fn endpoint ->
+      MappingGenerator.endpoint_segment()
+      |> bind(&interface_mapping(params, endpoint, &1))
+    end)
+    |> list_of(min_length: 1, max_length: @interface_max_mappings)
```

Comment:
I stand by my previous comment:

```suggestion
    MappingGenerator.mapping(params)
    |> list_of(min_length: 1, max_length: @interface_max_mappings)
```

one problem I see with this approach is that it makes it impossible to create endpoints with only one token, such as `/endpoint`.
While this is deprecated for object aggregates, it is perfectly fine for individual interfaces